config/v1/tests/authentications.config.openshift.io/ExternalOIDCWithUpstreamParity.yaml

```diff
@@ -0,0 +1,477 @@
+                prefix:
+                  prefixString: "myoidc:"
+
+    - name: Missing requiredClaim when type is RequiredClaim
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimValidationRules:
+              - type: RequiredClaim
+            claimMappings:
+              username:
+                claim: "preferred_username"
+                prefixPolicy: Prefix
+                prefix:
+                  prefixString: "myoidc:"
+      expectedError: "requiredClaim must be set when type is 'RequiredClaim'"
+
+    - name: Valid CEL expression configuration
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              username:
+                claim: "preferred_username"
+                prefixPolicy: Prefix
+                prefix:
+                  prefixString: "myoidc:"
+            claimValidationRules:
+              - type: CEL
+                cel:
+                  expression: "claims.email.endsWith('@example.com')"
+                  message: "email must be from example.com"
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimValidationRules:
+              - type: CEL
+                cel:
+                  expression: "claims.email.endsWith('@example.com')"
+                  message: "email must be from example.com"
+            claimMappings:
+              username:
+                claim: "preferred_username"
+                prefixPolicy: Prefix
+                prefix:
+                  prefixString: "myoidc:"
+
+    - name: Missing cel for CEL type
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              username:
+                claim: "preferred_username"
+                prefixPolicy: Prefix
+                prefix:
+                  prefixString: "myoidc:"
+            claimValidationRules:
+              - type: CEL
+      expectedError: "cel must be set when type is 'CEL', and forbidden otherwise"
+
+    - name: Empty expression in the CEL field
```

Comment:
Same comment as above regarding being more descriptive as to what field this test is for